src/server_manager/web_app/ui_components/app-root.js

```diff
@@ -658,16 +656,10 @@ export class AppRoot extends mixinBehaviors
   /**
    * Gets the ServerView for the server given by its id
    * @param {string} displayServerId
-   * @returns {ServerView}
+   * @returns {Promise<ServerView>}
    */
-  getServerView(displayServerId) {
-    if (!displayServerId) {
-      return null;
-    }
-    // Render to ensure that the server view has been added to the DOM.
-    this.$.serverView.querySelector('dom-repeat').render();
```

Comment:
Wow, thanks for finding the root cause. This seems like a very tricky bug.

There is still one thing that I don't understand. Why doesn't forcing a re-render address (side-step) the issue? I would expect it to rebind all of the views regardless of how the serverList items were added/removed.

Reply:
This is what happens:
serverList is:
- A
- B
- C

When you delete B, the dom for the server views will look like:
- id: A, properties A
- id: C, properties B

That's because the id, used to lookup the view, is bound to the serverList. But all the other properties are set directly on the components, bypassing the model binding. The rebinding works, for the properties that are bound. That's why I said we are doing it wrong. The entire state should be bound. That could be done with a DisplayServer kind of object that we bind.

We don't have the problem with the access keys because we use the AccessKeyRow for binding.

Reply:
Thanks, that clears things up.

What I take away from this is that we're mixing MVVM and MVP. Some properties are bound through the ViewModel (`ServerListEntry`) and others are bound directly via the presenter (`app.ts`). I think this would be great to flesh out when we refactor parts of the codebase into LitElement apps.

Reply:
For the most part we should be binding via the webcomponent model. Dealing with direct property setting and the views can tie us to a specific layout and is a pain.